src/lib/router.ts

```diff
@@ -1,20 +1,28 @@
-import type { ProjectsParams, Route, ProjectRoute } from "./router/definitions";
-import type { Readable } from "svelte/store";
+import type {
+  LoadedRoute,
+  ProjectRoute,
+  ProjectsParams,
+  Route,
+} from "@app/lib/router/definitions";
 
-import { get, writable, derived } from "svelte/store";
+import { get, writable } from "svelte/store";
+
+import * as mutexExecutor from "@app/lib/mutexExecutor";
+import { loadSeedRoute } from "@app/views/seeds/router";
 import { unreachable } from "@app/lib/utils";
 
 // This is only respected by Safari.
 const documentTitle = "Radicle Interface";
 
-export const historyStore = writable<Route[]>([{ resource: "home" }]);
+const LOADING_ROUTE: Extract<Route, { resource: "loading" }> = {
+  resource: "loading",
+  params: { url: window.location.toString() },
+};
+
+export const showLoadingBar = writable<boolean>(true);
```

Comment:
This makes sense. I initially had it in `LoadingBar.svelte` in the module context, because it's that component's state. But unfortunately it wasn't possible to import it into a .ts file due to the current toolchain limitations, so I moved it here.

What you're proposing makes sense though, because we can also view it as part of the router API. e1bf9e7855ac89f0cacd7a8829161feb88914182